### ISSUE TITLE
Need help understanding the CSS formats

### ISSUE THREAD

**Issue by knwpsk** [earlduque/Slack-Dark-Theme#31]
Hey, I've tried hard to figure this out on my own but can't.
In this theme (CSS), when I open a Thread in slack, the Sender Name background is much lighter (light gray?) than the surrounding background (almost black).  The names are hard to read, without better contrast.
I want to change the name background to be same as the surrounding background (which is true in the main chat window, but not in Thread window).
I can't find that name background property in the CSS file.  I've tried a few that didn't have any effect.  Any help?

And more generally, is there any guide to understanding the structure of the CSS file for Slack?
(I've also tried opening Slack in browser and using Inspect, but that still led me to several failed attempts)

**Comment by alinoancea** [comment]
Hey, have you tried opening Slack in developer mode?
```
export SLACK_DEVELOPER_MENU=true
slack
```
After that you can easily inspect every element like you would do in browser.

**Comment by knwpsk** [comment]
Alinoancea, thanks... but I don't understand.  Where would one do that ^ ?  "Export" isn't a command that Windows likes.

**Comment by alinoancea** [comment]
Sorry for that, the command above is for Unix. Try
```
set SLACK_DEVELOPER_MENU=1
<path_to_slack_directory>\slack.exe
```

This should open slack in developer mode.

**Comment by knwpsk** [comment]
NM I figured it out (how to run Slack windows app in Dev Mode).  For anyone else reading:

In Win 10:
- Start Menu > type "slack" > Right Click > More > Open File Location
(This takes you to the folder where you'll find the Slack shortcut)
- Highlight the shortcut > Right Click > Copy > Paste
(Creates a new copy of the shortcut)
In the Target field, paste this before the existing path:
C:\Windows\System32\cmd.exe /c " SET SLACK_DEVELOPER_MENU=TRUE && start 
Save.
Use this shortcut to open Slack in Dev mode.  Once open, type CTRL + ALT + I (letter "I") to open Inspector window.

**Comment by knwpsk** [comment]
OK so I got Slack open in dev mode.  Go to a thread.  Right click on a sender name and Inspect Element.  Dev Tools scrolls to the right place... but I still don't see a background color there.  Similarly, some Senders have a Status Icon next to their names, and that icon has the same light-gray background... but the inspector window doesn't show a Background Color element there, either.  
Any more help, pretty please?

(I'm not experienced with style sheets...)

**Comment by knwpsk** [comment]
Oh tg, I finally found it.

Line 8713 in the CSS file:  I changed color AAAAAA to 222:

span.c-message__sender.c-message_kit__sender {
    background: #222;

I never did find it in the Thread section; rather, I found it by inspecting the Sender in the main message window.  This makes no sense to me, because the Sender in the main window _doesn't_ have the light grey background.  But I found the element called Background for that, and changed it anyway, and that fixed the Thread window.  
If someone could kindly teach me why -- how this works -- I'd love to hear it.

>> I suggest this change should be merged into the dark.css file for this project (but without understanding better, I'm not going to touch it).

**Comment by knwpsk** [comment]
Here's a related question.
On the top of the Threads pane, there's an "X" icon to close the pane.  With dark.css installed, both the "X" and its background are flat black, so you can't see the X.  
While running in Dev mode, I use the element inspector (Dev Tools) to drill down to the section for that "X" element.  But there is no Foreground or Background entry to describe the colors for it.  
I assume it must inherit its colors from somewhere else... but that's where I don't know what to do.  How do I identify that inheritance?  etc.  
Thanks for any help!

**Comment by alinoancea** [comment]
I think that you are talking about the 'X' button which I tried to fix in #32. You can try it by adding 'https://raw.githubusercontent.com/alinoancea/Slack-Dark-Theme/incorrect-elements-in-downloads-tab/dark.css' as url in ssb-interop.js because it's not merge in master branch, yet.

**Comment by knwpsk** [comment]
Alinoancea, yes, the "X" in the Downloads pane looks like it's the same "X" in the Threads pane.  Thanks for that.

With the help of a colleague yesterday, we found the culprits, and I fixed them by adding these two items to the end of dark.css.  (I forked this project and added it to my own copy).  This addresses:
1. Search results were showing up with white background.
2. The "X" in the Downloads/Threads pane was black-on-black.

If someone wants to make these changes in the main branch, great.  I'm not touching it, I'm too new at style sheets.

.p-threads_view__default_background {
    background: #222;
}

.c-icon_button--light, .c-icon_button--light.c-button-unstyled {
    color: #a1a1a1;
}

**Comment by alinoancea** [comment]
No problem. You can try and play more with modifying css properties. For example after I've added you changes from master, I've spotted that the sender name is not so visible. Also when I hover the 'X' button the button color changes to black.
![image](https://user-images.githubusercontent.com/18450669/60982228-650b8900-a340-11e9-84b2-296359c04f37.png)

**Comment by knwpsk** [comment]
It's not the properties per se.  It's the structure of the css file and all of the inheritance that goes on.  Makes it very difficult to understand what might interact with what else.  I know I "fixed" the two things I named; but I can't be sure what else might be affected.  So I'd rather not "hack at it", but let someone more experienced see what I did and merge it if appropriate. 

(And thanks for being helpful!)
